lib/src/rules/no_leading_underscores_for_local_identifiers.dart

```diff
@@ -92,6 +96,20 @@ class _Visitor extends SimpleAstVisitor<void> {
     rule.reportLintForToken(id, arguments: [id.lexeme]);
   }
 
+  void checkPattern(DartPattern pattern) {
+    if (pattern is DeclaredVariablePattern) {
+      checkIdentifier(pattern.name);
+    } else if (pattern is LogicalAndPattern) {
+      // ?
+      // checkPattern(pattern.leftOperand);
+      checkPattern(pattern.rightOperand);
+    } else if (pattern is LogicalOrPattern) {
+      // ?
+      // checkPattern(pattern.leftOperand);
+      checkPattern(pattern.rightOperand);
```

Comment:
For a logical-or, in the absence of bugs, we only need to check one branch because all of the variables have to be defined on all branches (unless they're not referenced in the statements, in which case they're pointless). But I'd prefer to check the left operand so that the left-most declaration is the one being flagged.

Reply:
Perfect. Thanks!